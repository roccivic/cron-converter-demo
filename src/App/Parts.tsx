import { Part } from "./Part";
import { getUnits } from "cron-converter";

export const Parts = () => (
  <div style={{ display: "flex" }}>
    {getUnits().map((unit, index) => (
      <Part key={unit.name} id={unit.name} index={index} min={unit.min} max={unit.max} />
    ))}
  </div>
);
