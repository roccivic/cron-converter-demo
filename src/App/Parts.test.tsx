import { render } from "@testing-library/react";
import { Parts } from "./Parts";

jest.mock('./Part', () => ({
  Part: (props: any) => <div {...props}></div>
}))

test("<Parts />", () => {
  const comp = render(<Parts />);
  expect(comp.container).toMatchSnapshot();
});
