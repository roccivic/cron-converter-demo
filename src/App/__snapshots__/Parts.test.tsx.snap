// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Parts /> 1`] = `
<div>
  <div
    style="display: flex;"
  >
    <div
      id="minute"
      index="0"
      max="59"
      min="0"
    />
    <div
      id="hour"
      index="1"
      max="23"
      min="0"
    />
    <div
      id="day"
      index="2"
      max="31"
      min="1"
    />
    <div
      id="month"
      index="3"
      max="12"
      min="1"
    />
    <div
      id="weekday"
      index="4"
      max="6"
      min="0"
    />
  </div>
</div>
`;
